vspace/src/main/java/edu/asu/diging/vspace/core/factory/impl/SlideFactory.java

```diff
@@ -1,34 +1,49 @@
 package edu.asu.diging.vspace.core.factory.impl;
 
 import java.util.ArrayList;
+import java.util.List;
 
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import edu.asu.diging.vspace.core.factory.IChoiceFactory;
 import edu.asu.diging.vspace.core.factory.ISlideFactory;
+import edu.asu.diging.vspace.core.model.IBranchingPoint;
+import edu.asu.diging.vspace.core.model.IChoice;
 import edu.asu.diging.vspace.core.model.IContentBlock;
 import edu.asu.diging.vspace.core.model.IModule;
 import edu.asu.diging.vspace.core.model.ISlide;
+import edu.asu.diging.vspace.core.model.impl.BranchingPoint;
 import edu.asu.diging.vspace.core.model.impl.Slide;
 import edu.asu.diging.vspace.web.staff.forms.SlideForm;
 
 @Service
 public class SlideFactory implements ISlideFactory {
+    
+    @Autowired
+    private IChoiceFactory choiceFactory;
 
     /*
      * (non-Javadoc)
      * 
      * @see
-     * edu.asu.diging.vspace.core.factory.impl.ISlideFactory#createSlide(java.lang.
-     * String, java.lang.String)
+     * edu.asu.diging.vspace.core.factory.impl.ISlideFactory#createSlide(edu.asu.diging.vspace.core.model.IModule, edu.asu.diging.vspace.web.staff.forms.SlideForm)
      */
     @Override
-    public ISlide createSlide(IModule module, SlideForm form) {
-        ISlide slide = new Slide();
+    public ISlide createSlide(String type, IModule module, SlideForm form) {
         
-        slide.setContents(new ArrayList<IContentBlock>());
+        ISlide slide;
+        if(type.equals("Slide")) {
```

Comment:
do not hardcode strings. if there is no other way than doing it this way, use enums.